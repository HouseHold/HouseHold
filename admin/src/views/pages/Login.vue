<template>
  <CContainer class="d-flex align-items-center min-vh-100">
    <CRow class="justify-content-center">
      <CCol md="8">
        <CCardGroup>
          <CCard class="p-4">
            <CCardBody>
              <CForm>
                <h1>Login</h1>
                <p class="text-muted">Sign In to your account</p>
                <CInput
                  placeholder="Username"
                  autocomplete="username email"
                >
                  <template #prepend-content><CIcon name="cil-user"/></template>
                </CInput>
                <CInput
                  placeholder="Password"
                  type="password"
                  autocomplete="curent-password"
                >
                  <template #prepend-content><CIcon name="cil-lock-locked"/></template>
                </CInput>
                <CRow>
                  <CCol col="6">
                    <CButton color="primary" class="px-4">Login</CButton>
                  </CCol>
                  <CCol col="6" class="text-right">
                    <CButton color="link" class="px-0">Forgot password?</CButton>
                  </CCol>
                </CRow>
              </CForm>
            </CCardBody>
          </CCard>
          <CCard
            color="primary"
            text-color="white"
            class="text-center py-5 d-md-down-none"
            style="width:44%"
            body-wrapper
          >
            <h2>Sign up</h2>
            <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>
            <CButton
              color="primary"
              class="active mt-3"
            >
              Register Now!
            </CButton>
          </CCard>
        </CCardGroup>
      </CCol>
    </CRow>
  </CContainer>
</template>

<script>
export default {
  name: 'Login'
}
</script>
