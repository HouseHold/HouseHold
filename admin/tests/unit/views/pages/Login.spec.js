import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import CoreuiVue from '@coreui/vue'
import Login from '@/views/pages/Login'

Vue.use(CoreuiVue)

describe('Login.vue', () => {
  it('has a name', () => {
    expect(Login.name).toBe('Login')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Login)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Login', () => {
    const wrapper = shallowMount(Login)
    expect(wrapper.is(Login)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = shallowMount(Login)
    expect(wrapper.find('h1').text()).toMatch('Login')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Login)
    expect(wrapper.element).toMatchSnapshot()
  })
})
