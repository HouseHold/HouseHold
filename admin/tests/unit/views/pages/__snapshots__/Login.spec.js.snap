// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue renders correctly 1`] = `
<ccontainer-stub
  class="d-flex align-items-center min-vh-100"
  tag="div"
>
  <crow-stub
    class="justify-content-center"
    gutters="true"
    tag="div"
  >
    <ccol-stub
      md="8"
      tag="div"
    >
      <ccardgroup-stub
        tag="div"
      >
        <ccard-stub
          class="p-4"
        >
          <ccardbody-stub>
            <cform-stub>
              <h1>
                Login
              </h1>
               
              <p
                class="text-muted"
              >
                Sign In to your account
              </p>
               
              <cinput-stub
                autocomplete="username email"
                lazy="400"
                placeholder="Username"
                type="text"
              />
               
              <cinput-stub
                autocomplete="curent-password"
                lazy="400"
                placeholder="Password"
                type="password"
              />
               
              <crow-stub
                gutters="true"
                tag="div"
              >
                <ccol-stub
                  col="6"
                  tag="div"
                >
                  <cbutton-stub
                    activeclass="active"
                    class="px-4"
                    color="primary"
                    event="click"
                    exactactiveclass="active"
                    routertag="a"
                    target="_self"
                    type="button"
                  >
                    Login
                  </cbutton-stub>
                </ccol-stub>
                 
                <ccol-stub
                  class="text-right"
                  col="6"
                  tag="div"
                >
                  <cbutton-stub
                    activeclass="active"
                    class="px-0"
                    color="link"
                    event="click"
                    exactactiveclass="active"
                    routertag="a"
                    target="_self"
                    type="button"
                  >
                    Forgot password?
                  </cbutton-stub>
                </ccol-stub>
              </crow-stub>
            </cform-stub>
          </ccardbody-stub>
        </ccard-stub>
         
        <ccard-stub
          bodywrapper="true"
          class="text-center py-5 d-md-down-none"
          color="primary"
          textcolor="white"
        >
          <h2>
            Sign up
          </h2>
           
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
           
          <cbutton-stub
            activeclass="active"
            class="active mt-3"
            color="primary"
            event="click"
            exactactiveclass="active"
            routertag="a"
            target="_self"
            type="button"
          >
            
            Register Now!
          
          </cbutton-stub>
        </ccard-stub>
      </ccardgroup-stub>
    </ccol-stub>
  </crow-stub>
</ccontainer-stub>
`;
